src/Data/Time/PreciseDuration.purs

```diff
@@ -12,58 +14,85 @@ module Data.Time.PreciseDuration
   , toWeeks
   ) where
 
-import Prelude (($), (/), (<<<), (<>))
+import Prelude (($), (/), (>>>), (<<<), (<>))
 
+import Data.BigInt (BigInt)
 import Data.BigInt as BigInt
 import Data.Decimal (Decimal)
 import Data.Decimal as Decimal
 
-import Data.Time.PreciseDuration.Internal (PreciseDuration(..), bigIntToDecimal, day, hour, micro, milli, minute, second, toNanosecondsBI, toNanosecondsD, week)
-import Data.Time.PreciseDuration.Internal (PreciseDuration(..)) as PreciseDurationType
+import Data.Time.PreciseDuration.Internal (PreciseDuration(..), day, hour, micro, milli, minute, second, toNanosecondsD, week)
+import Data.Time.PreciseDuration.Internal (PreciseDuration) as PreciseDurationType
 
 toString :: PreciseDuration -> String
-toString =
-  case _ of
-    Nanoseconds d -> BigInt.toString d <> "ns"
-    Microseconds d -> Decimal.toString d <> "us"
-    Milliseconds d -> Decimal.toString d <> "ms"
-    Seconds d -> Decimal.toString d <> "s"
-    Minutes d -> Decimal.toString d <> "m"
-    Hours d -> Decimal.toString d <> "h"
-    Days d -> Decimal.toString d <> "d"
-    Weeks d -> Decimal.toString d <> "w"
+toString = case _ of
+  Nanoseconds d -> Decimal.toString d <> "ns"
+  Microseconds d -> Decimal.toString d <> "us"
+  Milliseconds d -> Decimal.toString d <> "ms"
+  Seconds d -> Decimal.toString d <> "s"
+  Minutes d -> Decimal.toString d <> "m"
+  Hours d -> Decimal.toString d <> "h"
+  Days d -> Decimal.toString d <> "d"
+  Weeks d -> Decimal.toString d <> "w"
+
+-- Smart constructors
+make = { nanoseconds, microseconds, milliseconds, seconds, minutes, hours, days, weeks }
```

Comment:
I don't understand the reason for constructing and exporting this record. We are already exporting all of these functions. Additionally it has no type signature.

Reply:
It's to make using the smart constructors less painful in the tests, so you don't have to enumerate all of them in the imports list

Reply:
Isn't it already imported qualified? We should just remove this. There's no utility in exporting this as a public API.

Reply:
ok